### PR TITLE
[docs] Add registry auth information to docker agent docs

### PR DESCRIPTION
**Pull request by benpankow**
## Summary

Starts to document examples of mounting Docker access creds into the Docker agent.

### CODE REVIEW

**Comment by benpankow** [comment]
Current dependencies on/for this PR:
* master
  * **PR #9409** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/9409" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/9409?utm_source=stack-comment).

File: docs/content/dagster-cloud/deployment/agents/docker.mdx [hunk 1]
```diff
@@ -102,6 +102,41 @@ height={320}
 
 ---
 
+## Granting Registry Access to Docker Agent
```
Comment (erinkcochran87) [inline docs/content/dagster-cloud/deployment/agents/docker.mdx:105]: Can you make this guy use sentence case, please?

File: docs/content/dagster-cloud/deployment/agents/docker.mdx [hunk 2]
```diff
@@ -102,6 +102,41 @@ height={320}
...
+
+The Docker agent runs Docker operations from within the `dagster-cloud-agent` container, including pulling user code images and running them as new containers. The `dagster-cloud-agent` container requires access to credentials in order to perform these operations. Depending on the registry provider you are using, authentication may look different. Often, credentials are stored in the Docker config file; `~/.docker/config.json`.
```
Comment (erinkcochran87) [inline docs/content/dagster-cloud/deployment/agents/docker.mdx:107]:
```suggestion
The Docker agent runs Docker operations from within the `dagster-cloud-agent` container, including pulling user code images and running them as new containers. The `dagster-cloud-agent` container requires access to credentials in order to perform these operations.

Depending on the registry provider you're using, authentication may look different. Credentials are often stored in the Docker config file (`~/.docker/config.json`).
```

- Splitting into two paragraphs to improve readability. I don't think this takes away from what's being said, but feel free to push back on this.
- Removed the semicolon preceding the name of the Docker file. As the file name itself isn't a complete sentence, this isn't needed

File: docs/content/dagster-cloud/deployment/agents/docker.mdx [hunk 3]
```diff
@@ -102,6 +102,41 @@ height={320}
...
+  --restart on-failure \
+  -it docker.io/dagster/dagster-cloud-agent:latest \
+  dagster-cloud agent run /opt/dagster/app
+```
+
+### Using Google Container Regsitry
+
+The easiest way to grant the agent container access to a GCR repository is to use the [standalone GCR credential helper](https://github.com/GoogleCloudPlatform/docker-credential-gcr) to create an access token:
+
+    # Replace $YOUR_REGISTRY with the GCR registry you are using, e.g. https://gcr.io
```
Comment (erinkcochran87) [inline docs/content/dagster-cloud/deployment/agents/docker.mdx:132]: This should be formatted the same way as other code blocks, like on line 117. Without it, the formatting looks funky when the site is built + rendered.

**Review by erinkcochran87 — COMMENTED** [review]
@benpankow After reading through the changes, it sounds like users should be granting registry permissions *before* they do anything else with the agent setup - is that correct?

If that's the case, I don't think the end of this guide is the right place for this section. To me, it feels like it was tacked on and makes the steps feel like they're out of order. Some thoughts on addressing this:

- Make the registry access step 1 and bump the other steps down (current step 1 becomes step 2, etc.)
- Fold this info into the prereqs section

**Comment by benpankow** [comment]
> users should be granting registry permissions _before_ they do anything else with the agent setup - is that correct?

Usually the setup should be done immediately before running the agent, e.g. in the same startup script that they run the agent (whether locally, in a VM etc). I've updated the PR structure to slot in this information earlier on which will hopefully make this more clear. Let me know what you think.

File: docs/content/dagster-cloud/deployment/agents/docker.mdx [hunk 4]
```diff
@@ -72,7 +72,29 @@ To complete the steps in this guide, you'll need:
 
 ---
 
-## Step 3: Start the agent
+## Step 3: Granting registry access to Docker agent
```
Comment (erinkcochran87) [inline docs/content/dagster-cloud/deployment/agents/docker.mdx:75]:
```suggestion
## Step 3: Grant registry access to the agent
```

Making this section heading consistent with other headings in the guide.

File: docs/content/dagster-cloud/deployment/agents/docker.mdx [hunk 5]
```diff
@@ -18,7 +18,7 @@ To complete the steps in this guide, you'll need:
 
 - **Permissions in Dagster Cloud that allow you to manage agent tokens**. Refer to the [User permissions documentation](/dagster-cloud/account/managing-users) for more info.
 - **To have Docker installed**
-- **Access to a container registry to which you can push images with Dagster code.** Additionally, your Docker agent must have the permissions required to pull images from the registry.
+- **Access to a container registry to which you can push images with Dagster code.** Additionally, your Docker agent must have the permissions required to pull images from the registry. See [below](#granting-registry-access-to-docker-agent) for more details.
```
Comment (erinkcochran87) [inline docs/content/dagster-cloud/deployment/agents/docker.mdx:21]:
```suggestion
- **Access to a container registry to which you can push images with Dagster code.** Additionally, your Docker agent must have the permissions required to pull images from the registry. See [below](#step-3-grant-registry-access-to-the-agent) for more details.
```

This fixes a broken link, updating it to use the new name of the section heading.

**Review by erinkcochran87 — COMMENTED** [review]
Two small changes (fixing a link and the section heading) but this otherwise looks good!

**Review by alangenfeld — CHANGES_REQUESTED** [review]
to your queue for recommended changes

**Pull request closed without merging**